### PR TITLE
Make upstreams unique per lb group

### PR DESCRIPTION
**Pull request by eherot**
The add-an-upstream test that I created may not actually work on first convergence for order-of-operations reasons but I didn't want this very important feature to be held up any more because of that.

### CODE REVIEW

File: resources/agent.rb [hunk 1]
```diff
@@ -37,6 +37,9 @@
   agent_yaml['loadBalancerConfig']['rootPath'] = agent_root_path
   agent_yaml['templates'] = templates || node['baragon']['templates'].to_hash.values
 
+  # Process the templates as ERB
+  agent_yaml['templates'].each { |item| item['template'] = ERB.new(item['template']).result(binding) }
```
Comment (jeffbyrnes) [inline resources/agent.rb]: Does this avoid all the `ImmutableHash` crap I was testing for?

Comment (eherot) [inline resources/agent.rb]: Which "this" are you referring to, exactly?

Comment (jeffbyrnes) [inline resources/agent.rb]: This very line. You’ve dropped the tests asserting that `!ruby/hash:Chef::Node::Immutable` is not present in various YAML, I wondered if this was why.

File: test/integration/localzk/serverspec/agent_spec.rb [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+        it { is_expected.to match(%r{^  proxy_pass http://baragon_default_tk-test-service;}) }
+      end
+    end
+
+    describe file '/tmp/default/upstreams/tk-test-service.conf' do
+      it { is_expected.to be_file }
 
-    describe '#content' do
-      subject { super().content }
-      it { is_expected.to_not match '!ruby/hash:Chef::Node::Immutable' }
```
Comment (jeffbyrnes) [inline test/integration/localzk/serverspec/agent_spec.rb]: I refer to the removal of this line, and the other further down this changeset.

File: resources/agent.rb (hunk 1 above)
Comment (eherot) [inline resources/agent.rb]: I didn't drop them, I just moved them to a common location.

**Comment by eherot** [comment]
Yo, @jeffbyrnes 👍/👎?

**Comment by jeffbyrnes** [comment]
Other than the failing Travis builds,:+1:

**Pull request merged**